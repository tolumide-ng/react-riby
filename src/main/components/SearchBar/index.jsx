import React, { useState } from 'react';
import { connect } from 'react-redux';
import { searchGifAction } from '../../../store/modules/searchGif/actions';
import Button from '../Button';

const SearchBar = ({
  history,
  searchGifStatus,
  searchGifs,
  searchText,
  setSearchText,
}) => {
  const handleChange = e => {
    e.preventDefault();
    const { value } = e.target;
    setSearchText(value);
  };

  const handleSubmit = e => {
    e.preventDefault();
    if (searchText.length > 1) {
      searchGifs({ history, searchText });
    }
  };

  return (
    <div className="w-full mt-20 flex items-center justify-center">
      <form
        onSubmit={handleSubmit}
        className="flex w-10/12 items-center justify-center items-center"
      >
        <input
          value={searchText}
          onChange={handleChange}
          className="border w-7/12 px-2 py-1"
          placeholder="search here"
          data-testid="inputBox"
        />
        <Button
          type="submit"
          title="Search"
          disabled={searchGifStatus === 'pending'}
          classes={`text-white px-2 py-1 ${
            searchGifStatus === 'pending'
              ? 'bg-yellow-600'
              : 'bg-blue-500 '
          }`}
        />
      </form>
    </div>
  );
};

const mapStateToProps = state => ({
  searchResult: state.searchGifReducer.searchGifResult,
  searchGifStatus: state.searchGifReducer.searchGifStatus,
  searchError: state.searchGifReducer.searchError,
});

const mapDispatchToProps = dispatch => ({
  searchGifs: ({ history, searchText }) =>
    dispatch(searchGifAction({ history, searchText })),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(SearchBar);
