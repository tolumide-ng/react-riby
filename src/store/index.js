import { createStore, applyMiddleware } from 'redux';
import ReduxThunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import rootReducer from './modules';

const middlewares = applyMiddleware(ReduxThunk);

const store = createStore(rootReducer, composeWithDevTools);

export default store;
