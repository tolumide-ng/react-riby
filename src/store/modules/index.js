import { combineReducers } from 'redux';
import allGifReducer from './allGifs/reducers';
import detailedGifReducer from './detailedGif/reducers';
import searchGifReducer from './searchGif/reducers';

const rootReducer = combineReducers({
  allGifReducer,
  detailedGifReducer,
  searchGifReducer,
});

export default rootReducer;
