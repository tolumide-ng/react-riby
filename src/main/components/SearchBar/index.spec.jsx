import { render, fireEvent, cleanup } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import SearchBar from './index';
import store from '../../../store';

const history = { push: jest.fn() };
const searchGifStatus = '';
const setSearchText = () => jest.fn();
const searchText = '';
const searchGifs = () => jest.fn();

describe('SearchBar', () => {
  test('Should return the Search Bar', () => {
    const defaultProps = {
      history,
      searchGifStatus,
      setSearchText,
      searchText,
      searchGifs,
    };
    const placeholderName = 'search here';
    const { asFragment, getByTestId, getByPlaceholderText } = render(
      <Provider store={store}>
        <MemoryRouter>
          <SearchBar {...defaultProps} />
        </MemoryRouter>
      </Provider>,
    );

    expect(asFragment).toMatchSnapshot();
    const input = getByPlaceholderText(placeholderName);
    const inputBox = getByTestId('inputBox');
    fireEvent.change(inputBox, { target: { value: '' } });
    fireEvent.blur(inputBox);

    expect(searchGifs()).not.toHaveBeenCalled();
  });
});
