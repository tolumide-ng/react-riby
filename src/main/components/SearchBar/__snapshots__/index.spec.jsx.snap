// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Should return the Search Bar 1`] = `[Function]`;
